lambdas/build-cri-oauth-request/src/main/java/uk/gov/di/ipv/core/buildcrioauthrequest/BuildCriOauthRequestHandler.java

```diff
@@ -96,6 +102,10 @@ public class BuildCriOauthRequestHandler
     private final ClientOAuthSessionDetailsService clientOAuthSessionDetailsService;
     private final Gpg45ProfileEvaluator gpg45ProfileEvaluator;
     private final SessionCredentialsService sessionCredentialsService;
+    private final CriApiService criApiService;
+
+    private static final Integer DEFAULT_OAUTH_CRI_ENC_KEY_CACHE_DURATION_MINUTES = 5;
+    private static CachedOAuthCriEncryptionKey cachedOAuthCriEncryptionKey;
```

Comment:
Careful - although in dev/build all our CRIs use the same keys, in production they won't!

We'll probably need to store a cache indexed by CRI + connection id

Reply:
oops good point!